service/create/validation.go

```diff
@@ -3,17 +3,31 @@ package create
 import (
 	"github.com/giantswarm/awstpr"
 	"github.com/giantswarm/awstpr/spec/aws"
+	"github.com/giantswarm/clustertpr/spec"
 	"github.com/giantswarm/microerror"
 )
 
-func validateWorkers(workers []aws.Node) error {
-	if len(workers) < 1 {
+func validateMasters(awsMasters []aws.Node, masters []spec.Node) error {
+	// Currently only a single master is expected.
+	if len(awsMasters) != 1 || len(masters) != 1 {
+		return microerror.Mask(invalidMasterNodeCountError)
+	}
+
+	return nil
+}
+
+func validateWorkers(awsWorkers []aws.Node, workers []spec.Node) error {
```

Comment:
validateWorkers is extended to check the workers and AWS workers blocks contain the same number of nodes.